packages/flutter_tools/lib/src/fuchsia/fuchsia_kernel_compiler.dart

```diff
@@ -103,4 +91,48 @@ class FuchsiaKernelCompiler {
       throwToolExit('Build process failed');
     }
   }
+
+  /// Provide flags that are affected by [BuildInfo]
+  List<String> getBuildInfoFlags({
```

Comment:
I would make this method `static` and annotate it with `@visibleForTesting` from `package:meta/meta.dart`